### DISCUSSION TITLE
Relationship existence: Error in this documentation example?

### DISCUSSION THREAD

**Discussion by LeCoupa** [adonisjs/core#2137] [Help] [answered]
I think there is an issue with the example below:
https://preview.adonisjs.com/guides/model-relations/introduction#querying-relationship-existence

```
const veterans = await User
  .query()
  .whereHas('courses', (query) => {
    query.wherePivot('completion_percentage', 100)
  }, '>', 2)
```

TypeScript seems to stay that `whereHas()` cannot accept 4 arguments.

**Comment by thetutlage** [comment, marked as answer]
Cannot suggest anything without looking at the exact Typescript error. The code does show that 4 arguments are allowed https://github.com/adonisjs/lucid/blob/develop/adonis-typings/relations.ts#L963-L968

**Reply by LeCoupa** [reply]
Sorry about this, I can't replicate the situation I had.
